pkg/internal/controller/controller_test.go

```diff
@@ -155,6 +155,152 @@ var _ = Describe("controller", func() {
 			Expect(err).To(Equal(context.DeadlineExceeded))
 		})
 
+		Context("prometheus metric reconcile_timeouts", func() {
+			var reconcileTimeouts dto.Metric
+
+			BeforeEach(func() {
+				ctrlmetrics.ReconcileTimeouts.Reset()
+				reconcileTimeouts.Reset()
+				ctrl.Name = testControllerName
+			})
+
+			It("should increment when ReconciliationTimeout fires", func(ctx SpecContext) {
+				Expect(func() error {
+					Expect(ctrlmetrics.ReconcileTimeouts.WithLabelValues(ctrl.Name).Write(&reconcileTimeouts)).To(Succeed())
+					if reconcileTimeouts.GetCounter().GetValue() != 0.0 {
+						return fmt.Errorf("metric reconcile timeouts not reset")
+					}
+					return nil
+				}()).Should(Succeed())
```

Comment:
Should this not be part of the before each?

Reply:
Yep good point - fixed and also rewrote to avoid the nesting